src/services/api/errors.ts

```diff
@@ -0,0 +1,41 @@
+//  TODO: use functions instead of classes to respect eslint, see  https://dev.to/damxipo/custom-exceptions-with-js-3aoc
+
+/**
+ * The api has answered with an error description.
+ */
+class ApiError extends Error {
+  public code: number;
+
+  constructor(message, statusCode = 500) {
+    super(message);
+    this.name = 'ApiError';
+    this.code = statusCode;
+    Object.setPrototypeOf(this, ApiError.prototype);
+    Error.captureStackTrace(this, ApiError);
```

Comment:
I am not sure if we need this, calling `super()` seems to be sufficient. I will remove  those calls and get back to stack tracing if we need to.